<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="A set of routines to quickly download and import the HGNC
    data set on mapping of gene symbols to gene entries in other popular
    databases or resources.">
<title>Download and Import the HUGO Gene Nomenclature Committee (HGNC) Data Set into R â€¢ hgnc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Download and Import the HUGO Gene Nomenclature Committee (HGNC) Data Set into R">
<meta property="og:description" content="A set of routines to quickly download and import the HGNC
    data set on mapping of gene symbols to gene entries in other popular
    databases or resources.">
<meta property="og:image" content="https://rmagno.eu/hgnc/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">hgnc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ramiromagno/hgnc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.svg" class="logo" alt=""><h1 id="hgnc-">hgnc <a class="anchor" aria-label="anchor" href="#hgnc-"></a>
</h1>
</div>
<!-- badges: start -->

<p>The <a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature Committee (HGNC)</a> is a committee of the Human Genome Organisation (HUGO) that sets the standards for human gene nomenclature.</p>
<p>The HGNC approves a unique and meaningful name for every known human gene, based on a group of experts. In addition, the HGNC also provides the mapping between gene symbols to gene entries in other popular databases or resources: the HGNC complete gene set.</p>
<p>The goal of <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> is to easily download and import the latest HGNC complete gene data set into R.</p>
<p>This data set provides a useful mapping of HGNC symbols to gene entries in other popular databases or resources, such as, the Entrez gene identifier or the UCSC gene identifier, among many others. Check the documentation of the function <code><a href="reference/import_hgnc_dataset.html">import_hgnc_dataset()</a></code> for a description of the several fields available.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> from CRAN:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"hgnc"</span><span class="op">)</span></span></code></pre></div>
<p>You can install the development version of <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> like so:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ramiromagno/hgnc"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="basic-usage">Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h3>
<p>To import the latest HGNC gene data set in tabular format directly into memory as a tibble do as follows:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Date of HGNC last update</span></span>
<span><span class="fu"><a href="reference/last_update.html">last_update</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2023-10-30 03:31:41 UTC"</span></span>
<span></span>
<span><span class="co"># Set the HGNC archive file to use for the remainder of the R-session</span></span>
<span><span class="fu"><a href="reference/use_hgnc_file.html">use_hgnc_file</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="reference/latest_archive_url.html">latest_archive_url</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; using hgnc file: https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/tsv/hgnc_complete_set.txt</span></span>
<span></span>
<span><span class="co"># Import the data set in tidy tabular format</span></span>
<span><span class="co"># NB: Multiple-value columns are kept as list-columns</span></span>
<span><span class="va">hgnc_dataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/import_hgnc_dataset.html">import_hgnc_dataset</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">hgnc_dataset</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 43,736</span></span>
<span><span class="co">#&gt; Columns: 55</span></span>
<span><span class="co">#&gt; $ hgnc_id                  &lt;chr&gt; "HGNC:5", "HGNC:37133", "HGNC:24086", "HGNC:7â€¦</span></span>
<span><span class="co">#&gt; $ hgnc_id2                 &lt;int&gt; 5, 37133, 24086, 7, 27057, 23336, 41022, 4152â€¦</span></span>
<span><span class="co">#&gt; $ symbol                   &lt;chr&gt; "A1BG", "A1BG-AS1", "A1CF", "A2M", "A2M-AS1",â€¦</span></span>
<span><span class="co">#&gt; $ name                     &lt;chr&gt; "alpha-1-B glycoprotein", "A1BG antisense RNAâ€¦</span></span>
<span><span class="co">#&gt; $ locus_group              &lt;chr&gt; "protein-coding gene", "non-coding RNA", "proâ€¦</span></span>
<span><span class="co">#&gt; $ locus_type               &lt;chr&gt; "gene with protein product", "RNA, long non-câ€¦</span></span>
<span><span class="co">#&gt; $ status                   &lt;chr&gt; "Approved", "Approved", "Approved", "Approvedâ€¦</span></span>
<span><span class="co">#&gt; $ location                 &lt;chr&gt; "19q13.43", "19q13.43", "10q11.23", "12p13.31â€¦</span></span>
<span><span class="co">#&gt; $ location_sortable        &lt;chr&gt; "19q13.43", "19q13.43", "10q11.23", "12p13.31â€¦</span></span>
<span><span class="co">#&gt; $ alias_symbol             &lt;list&gt; NA, "FLJ23569", &lt;"ACF", "ASP", "ACF64", "ACFâ€¦</span></span>
<span><span class="co">#&gt; $ alias_name               &lt;list&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, &lt;"iGb3 sâ€¦</span></span>
<span><span class="co">#&gt; $ prev_symbol              &lt;list&gt; NA, &lt;"NCRNA00181", "A1BGAS", "A1BG-AS"&gt;, NA,â€¦</span></span>
<span><span class="co">#&gt; $ prev_name                &lt;list&gt; NA, &lt;"non-protein coding RNA 181", "A1BG antâ€¦</span></span>
<span><span class="co">#&gt; $ gene_group               &lt;list&gt; "Immunoglobulin like domain containing", "Anâ€¦</span></span>
<span><span class="co">#&gt; $ gene_group_id            &lt;list&gt; "594", "1987", "725", "2148", "1987", "2148"â€¦</span></span>
<span><span class="co">#&gt; $ date_approved_reserved   &lt;date&gt; 1989-06-30, 2009-07-20, 2007-11-23, 1986-01-â€¦</span></span>
<span><span class="co">#&gt; $ date_symbol_changed      &lt;date&gt; NA, 2010-11-25, NA, NA, NA, 2005-09-01, NA, â€¦</span></span>
<span><span class="co">#&gt; $ date_name_changed        &lt;date&gt; NA, 2012-08-15, NA, NA, 2018-03-21, 2016-03-â€¦</span></span>
<span><span class="co">#&gt; $ date_modified            &lt;date&gt; 2023-01-20, 2013-06-27, 2023-01-20, 2023-01-â€¦</span></span>
<span><span class="co">#&gt; $ entrez_id                &lt;int&gt; 1, 503538, 29974, 2, 144571, 144568, 10087410â€¦</span></span>
<span><span class="co">#&gt; $ ensembl_gene_id          &lt;chr&gt; "ENSG00000121410", "ENSG00000268895", "ENSG00â€¦</span></span>
<span><span class="co">#&gt; $ vega_id                  &lt;chr&gt; "OTTHUMG00000183507", "OTTHUMG00000183508", "â€¦</span></span>
<span><span class="co">#&gt; $ ucsc_id                  &lt;chr&gt; "uc002qsd.5", "uc002qse.3", "uc057tgv.1", "ucâ€¦</span></span>
<span><span class="co">#&gt; $ ena                      &lt;list&gt; NA, "BC040926", "AF271790", &lt;"BX647329", "X6â€¦</span></span>
<span><span class="co">#&gt; $ refseq_accession         &lt;list&gt; "NM_130786", "NR_015380", "NM_014576", "NM_0â€¦</span></span>
<span><span class="co">#&gt; $ ccds_id                  &lt;list&gt; "CCDS12976", NA, &lt;"CCDS7242", "CCDS7241", "Câ€¦</span></span>
<span><span class="co">#&gt; $ uniprot_ids              &lt;list&gt; "P04217", NA, "Q9NQ94", "P01023", NA, "A8K2Uâ€¦</span></span>
<span><span class="co">#&gt; $ pubmed_id                &lt;list&gt; "2591067", NA, &lt;"11815617", "11072063"&gt;, &lt;"2â€¦</span></span>
<span><span class="co">#&gt; $ mgd_id                   &lt;list&gt; "MGI:2152878", NA, "MGI:1917115", "MGI:24491â€¦</span></span>
<span><span class="co">#&gt; $ rgd_id                   &lt;list&gt; "RGD:69417", NA, "RGD:619834", "RGD:2004", Nâ€¦</span></span>
<span><span class="co">#&gt; $ lsdb                     &lt;chr&gt; NA, NA, NA, "LRG_591|http://ftp.ebi.ac.uk/pubâ€¦</span></span>
<span><span class="co">#&gt; $ cosmic                   &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ omim_id                  &lt;list&gt; "138670", NA, "618199", "103950", NA, "61062â€¦</span></span>
<span><span class="co">#&gt; $ mirbase                  &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ homeodb                  &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ snornabase               &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ bioparadigms_slc         &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ orphanet                 &lt;chr&gt; NA, NA, NA, NA, NA, "410627", NA, NA, NA, NA,â€¦</span></span>
<span><span class="co">#&gt; $ pseudogene.org           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, "PGOHUM000002â€¦</span></span>
<span><span class="co">#&gt; $ horde_id                 &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ merops                   &lt;chr&gt; "I43.950", NA, NA, "I39.001", NA, "I39.007", â€¦</span></span>
<span><span class="co">#&gt; $ imgt                     &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ iuphar                   &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ kznf_gene_catalog        &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ `mamit-trnadb`           &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ cd                       &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ lncrnadb                 &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ enzyme_id                &lt;list&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, "2.4â€¦</span></span>
<span><span class="co">#&gt; $ intermediate_filament_db &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ rna_central_ids          &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ lncipedia                &lt;chr&gt; NA, "A1BG-AS1", NA, NA, "A2M-AS1", NA, "A2ML1â€¦</span></span>
<span><span class="co">#&gt; $ gtrnadb                  &lt;chr&gt; NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, Nâ€¦</span></span>
<span><span class="co">#&gt; $ agr                      &lt;chr&gt; "HGNC:5", "HGNC:37133", "HGNC:24086", "HGNC:7â€¦</span></span>
<span><span class="co">#&gt; $ mane_select              &lt;list&gt; &lt;"ENST00000263100.8", "NM_130786.4"&gt;, NA, &lt;"â€¦</span></span>
<span><span class="co">#&gt; $ gencc                    &lt;chr&gt; NA, NA, NA, "HGNC:7", NA, "HGNC:23336", NA, Nâ€¦</span></span></code></pre></div>
<p>The original data set does not contain the column <code>hgnc_id2</code>, which is added as a convenience by <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a>; this is because although the HGNC identifiers should formally contain the prefix <code>"HGNC:"</code>, it is often found elsewhere that they are stripped of this prefix, so the column <code>hgnc_id2</code> is also provided whose values only contain the integer part.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'hgnc_id'</span>, <span class="st">'hgnc_id2'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 43,736 Ã— 2</span></span>
<span><span class="co">#&gt;    hgnc_id    hgnc_id2</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1 HGNC:5            5</span></span>
<span><span class="co">#&gt;  2 HGNC:37133    37133</span></span>
<span><span class="co">#&gt;  3 HGNC:24086    24086</span></span>
<span><span class="co">#&gt;  4 HGNC:7            7</span></span>
<span><span class="co">#&gt;  5 HGNC:27057    27057</span></span>
<span><span class="co">#&gt;  6 HGNC:23336    23336</span></span>
<span><span class="co">#&gt;  7 HGNC:41022    41022</span></span>
<span><span class="co">#&gt;  8 HGNC:41523    41523</span></span>
<span><span class="co">#&gt;  9 HGNC:8            8</span></span>
<span><span class="co">#&gt; 10 HGNC:30005    30005</span></span>
<span><span class="co">#&gt; # â„¹ 43,726 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="locus-groups">Locus groups<a class="anchor" aria-label="anchor" href="#locus-groups"></a>
</h3>
<p>The HGNC defines a group name (<code>locus_group</code>) for a set of related locus types. Hereâ€™s how you can quickly check how many gene entries there are per locus group.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">locus_group</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 4 Ã— 2</span></span>
<span><span class="co">#&gt;   locus_group             n</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;               &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 protein-coding gene 19278</span></span>
<span><span class="co">#&gt; 2 pseudogene          14376</span></span>
<span><span class="co">#&gt; 3 non-coding RNA       9091</span></span>
<span><span class="co">#&gt; 4 other                 991</span></span></code></pre></div>
<p><code>locus_type</code> provides a finer classification:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">locus_group</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">locus_type</span>, sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">locus_group</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span>n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 23 Ã— 3</span></span>
<span><span class="co">#&gt; # Groups:   locus_group [4]</span></span>
<span><span class="co">#&gt;    locus_group         locus_type                     n</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;               &lt;chr&gt;                      &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1 non-coding RNA      RNA, long non-coding        5754</span></span>
<span><span class="co">#&gt;  2 non-coding RNA      RNA, micro                  1912</span></span>
<span><span class="co">#&gt;  3 non-coding RNA      RNA, transfer                591</span></span>
<span><span class="co">#&gt;  4 non-coding RNA      RNA, small nucleolar         568</span></span>
<span><span class="co">#&gt;  5 non-coding RNA      RNA, cluster                 119</span></span>
<span><span class="co">#&gt;  6 non-coding RNA      RNA, ribosomal                60</span></span>
<span><span class="co">#&gt;  7 non-coding RNA      RNA, small nuclear            50</span></span>
<span><span class="co">#&gt;  8 non-coding RNA      RNA, misc                     29</span></span>
<span><span class="co">#&gt;  9 non-coding RNA      RNA, Y                         4</span></span>
<span><span class="co">#&gt; 10 non-coding RNA      RNA, vault                     4</span></span>
<span><span class="co">#&gt; 11 other               immunoglobulin gene          230</span></span>
<span><span class="co">#&gt; 12 other               T cell receptor gene         206</span></span>
<span><span class="co">#&gt; 13 other               readthrough                  147</span></span>
<span><span class="co">#&gt; 14 other               fragile site                 116</span></span>
<span><span class="co">#&gt; 15 other               endogenous retrovirus        109</span></span>
<span><span class="co">#&gt; 16 other               complex locus constituent     69</span></span>
<span><span class="co">#&gt; 17 other               unknown                       68</span></span>
<span><span class="co">#&gt; 18 other               region                        38</span></span>
<span><span class="co">#&gt; 19 other               virus integration site         8</span></span>
<span><span class="co">#&gt; 20 protein-coding gene gene with protein product  19278</span></span>
<span><span class="co">#&gt; 21 pseudogene          pseudogene                 14136</span></span>
<span><span class="co">#&gt; 22 pseudogene          immunoglobulin pseudogene    203</span></span>
<span><span class="co">#&gt; 23 pseudogene          T cell receptor pseudogene    37</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setting-the-archive-version">Setting the archive version<a class="anchor" aria-label="anchor" href="#setting-the-archive-version"></a>
</h3>
<p>By default <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> will use the latest version of HGNC archive dataset, as returned by the function <code><a href="reference/latest_archive_url.html">latest_archive_url()</a></code>. Besides the latest archive, the <a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature Committee (HGNC)</a> website also provides monthly and quarterly updates. You can conveniently get the latest monthly and quarterly updates by running <code><a href="reference/latest_monthly_url.html">latest_monthly_url()</a></code> or <code><a href="reference/latest_quarterly_url.html">latest_quarterly_url()</a></code>. Use <code><a href="reference/list_archives.html">list_archives()</a></code> to list the all currently available for download archives. The column <code>url</code> contains the direct download link that you can pass to <code><a href="reference/use_hgnc_file.html">use_hgnc_file()</a></code> and <code><a href="reference/import_hgnc_dataset.html">import_hgnc_dataset()</a></code> to import the data into R.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/list_archives.html">list_archives</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 182 Ã— 7</span></span>
<span><span class="co">#&gt;    series  dataset           file     date       size  last_modified       url  </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;             &lt;chr&gt;    &lt;date&gt;     &lt;chr&gt; &lt;dttm&gt;              &lt;chr&gt;</span></span>
<span><span class="co">#&gt;  1 monthly hgnc_complete_set hgnc_coâ€¦ 2021-03-01 14M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  2 monthly hgnc_complete_set hgnc_coâ€¦ 2021-04-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  3 monthly hgnc_complete_set hgnc_coâ€¦ 2021-05-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  4 monthly hgnc_complete_set hgnc_coâ€¦ 2021-06-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  5 monthly hgnc_complete_set hgnc_coâ€¦ 2021-07-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  6 monthly hgnc_complete_set hgnc_coâ€¦ 2021-08-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  7 monthly hgnc_complete_set hgnc_coâ€¦ 2021-09-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  8 monthly hgnc_complete_set hgnc_coâ€¦ 2021-10-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt;  9 monthly hgnc_complete_set hgnc_coâ€¦ 2021-11-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt; 10 monthly hgnc_complete_set hgnc_coâ€¦ 2021-12-01 15M   2023-05-01 00:05:00 httpâ€¦</span></span>
<span><span class="co">#&gt; # â„¹ 172 more rows</span></span>
<span></span>
<span><span class="fu"><a href="reference/use_hgnc_file.html">use_hgnc_file</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="reference/latest_monthly_url.html">latest_monthly_url</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; using hgnc file: https://ftp.ebi.ac.uk/pub/databases/genenames/hgnc/archive/monthly/tsv/hgnc_complete_set_2023-10-01.txt</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="downloading-to-disk">Downloading to disk<a class="anchor" aria-label="anchor" href="#downloading-to-disk"></a>
</h3>
<p>If you prefer to download the data set as a file to disk first, you can use <code><a href="reference/download_archive.html">download_archive()</a></code>. Then, you can use <code><a href="reference/import_hgnc_dataset.html">import_hgnc_dataset()</a></code> to import the downloaded file into R.</p>
</div>
<div class="section level3">
<h3 id="convert-between-gene-identifiers">Convert between gene identifiers<a class="anchor" aria-label="anchor" href="#convert-between-gene-identifiers"></a>
</h3>
<p>Two convenience functions are provided to convert between different gene identifiers in the HGNC dataset: <code>hgnc_join</code> takes as input a data frame and adds an additional column from the HGNC dataset, and <code>hgnc_convert</code> converts a vector from one identifier to another.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>hgnc_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGNC:44948"</span>, <span class="st">"HGNC:43240"</span>, <span class="st">"HGNC:23357"</span>, <span class="st">"HGNC:1855"</span>, <span class="st">"HGNC:39400"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="co"># add in gene symbol</span></span>
<span>  <span class="fu"><a href="reference/hgnc_join.html">hgnc_join</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'hgnc_id'</span>, column <span class="op">=</span> <span class="st">'symbol'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="co"># add in entrez_id</span></span>
<span>  <span class="fu"><a href="reference/hgnc_join.html">hgnc_join</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'hgnc_id'</span>, column <span class="op">=</span> <span class="st">'entrez_id'</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 5 Ã— 3</span></span>
<span><span class="co">#&gt;   hgnc_id    symbol    entrez_id</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;         &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 HGNC:44948 GOLGA2P6     729668</span></span>
<span><span class="co">#&gt; 2 HGNC:43240 RNA5SP340 100873603</span></span>
<span><span class="co">#&gt; 3 HGNC:23357 MCTS1         28985</span></span>
<span><span class="co">#&gt; 4 HGNC:1855  CENPCP1        1061</span></span>
<span><span class="co">#&gt; 5 HGNC:39400 ASNSP4    100419423</span></span>
<span></span>
<span><span class="co"># convert a set of hgnc_ids to symbols</span></span>
<span><span class="va">hgnc_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HGNC:44948"</span>, <span class="st">"HGNC:43240"</span>, <span class="st">"HGNC:23357"</span>, <span class="st">"HGNC:1855"</span>, <span class="st">"HGNC:39400"</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/hgnc_convert.html">hgnc_convert</a></span><span class="op">(</span><span class="va">hgnc_ids</span>, from <span class="op">=</span> <span class="st">'hgnc_id'</span>, to <span class="op">=</span> <span class="st">'symbol'</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GOLGA2P6"  "RNA5SP340" "MCTS1"     "CENPCP1"   "ASNSP4"</span></span>
<span></span>
<span><span class="co"># convert a set of entrez_ids to ensembl_gene_ids</span></span>
<span><span class="va">entrez_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">79933</span>, <span class="fl">109623458</span>, <span class="fl">158471</span>, <span class="fl">54987</span>, <span class="fl">81631</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/hgnc_convert.html">hgnc_convert</a></span><span class="op">(</span><span class="va">entrez_ids</span>, from <span class="op">=</span> <span class="st">'entrez_id'</span>, to <span class="op">=</span> <span class="st">'ensembl_gene_id'</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "ENSG00000166317" NA                "ENSG00000106772" "ENSG00000162384"</span></span>
<span><span class="co">#&gt; [5] "ENSG00000140941"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="caching">Caching<a class="anchor" aria-label="anchor" href="#caching"></a>
</h3>
<p>By default <a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> will use memory-based caching through the package <a href="https://memoise.r-lib.org" class="external-link">memoise</a>. This will ensure that time consuming downloads are not rerun unnecessarily during a single R-session. Persistent disk based caching can also be enabled using the function <code>use_cache_dir</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/use_cache_dir.html">use_cache_dir</a></span><span class="op">(</span>cache_dir <span class="op">=</span>  <span class="st">'./hgnc_cache'</span><span class="op">)</span></span>
<span><span class="co">#&gt; using hgnc cache dir: ./hgnc_cache</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h2>
<p>You could go to <a href="https://www.genenames.org" class="external-link">www.genenames.org</a> and download the files yourself. So why the need for this R package?</p>
<p><a href="https://github.com/ramiromagno/hgnc" class="external-link">hgnc</a> really is just a convenience package. The main advantage is that the function <code><a href="reference/import_hgnc_dataset.html">import_hgnc_dataset()</a></code> reads in the data in tabular format with all the columns with the appropriate type (so you donâ€™t have to specify it yourself). As an extra step, those variables that contain multiple values are encoded as list-columns.</p>
<p>Remember that list-columns can be expanded with <code><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">tidyr::unnest()</a></code>. E.g., <code>alias_symbol</code> is a list-column containing multiple alternative aliases to the standard <code>symbol</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">'TP53'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'symbol'</span>, <span class="st">'alias_symbol'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 1 Ã— 2</span></span>
<span><span class="co">#&gt;   symbol alias_symbol</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;list&gt;      </span></span>
<span><span class="co">#&gt; 1 TP53   &lt;chr [2]&gt;</span></span>
<span></span>
<span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">'TP53'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'symbol'</span>, <span class="st">'alias_symbol'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">unnest</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="st">'alias_symbol'</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 2 Ã— 2</span></span>
<span><span class="co">#&gt;   symbol alias_symbol</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;       </span></span>
<span><span class="co">#&gt; 1 TP53   p53         </span></span>
<span><span class="co">#&gt; 2 TP53   LFS1</span></span></code></pre></div>
<p>In addition, we also provide the function <code><a href="reference/filter_by_keyword.html">filter_by_keyword()</a></code> that allows filtering the data set based on a keyword or regular expression. By default this function will look into all columns that contain gene symbols or names (<code>symbol</code>, <code>name</code>, <code>alias_symbol</code>, <code>alias_name</code>, <code>prev_symbol</code> and <code>prev_name</code>). It works automatically with list-columns too.</p>
<p>Look for entries in the data set that contain the keyword <code>"TP53"</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/filter_by_keyword.html">filter_by_keyword</a></span><span class="op">(</span><span class="st">'TP53'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 66 Ã— 4</span></span>
<span><span class="co">#&gt;    hgnc_id    hgnc_id2 symbol      name                                         </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;                                        </span></span>
<span><span class="co">#&gt;  1 HGNC:49685    49685 ABHD15-AS1  ABHD15 antisense RNA 1                       </span></span>
<span><span class="co">#&gt;  2 HGNC:20679    20679 ANO9        anoctamin 9                                  </span></span>
<span><span class="co">#&gt;  3 HGNC:40093    40093 BCAR3-AS1   BCAR3 antisense RNA 1                        </span></span>
<span><span class="co">#&gt;  4 HGNC:13276    13276 EI24        EI24 autophagy associated transmembrane protâ€¦</span></span>
<span><span class="co">#&gt;  5 HGNC:3345      3345 ENC1        ectodermal-neural cortex 1                   </span></span>
<span><span class="co">#&gt;  6 HGNC:27919    27919 ERVMER61-1  endogenous retrovirus group MER61 member 1   </span></span>
<span><span class="co">#&gt;  7 HGNC:56226    56226 FAM169A-AS1 FAM169A antisense RNA 1                      </span></span>
<span><span class="co">#&gt;  8 HGNC:4136      4136 GAMT        guanidinoacetate N-methyltransferase         </span></span>
<span><span class="co">#&gt;  9 HGNC:54868    54868 KLRK1-AS1   KLRK1 antisense RNA 1                        </span></span>
<span><span class="co">#&gt; 10 HGNC:6568      6568 LGALS7      galectin 7                                   </span></span>
<span><span class="co">#&gt; # â„¹ 56 more rows</span></span></code></pre></div>
<p>Restrict the search to the <code>symbol</code> column:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/filter_by_keyword.html">filter_by_keyword</a></span><span class="op">(</span><span class="st">'TP53'</span>, cols <span class="op">=</span> <span class="st">'symbol'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 23 Ã— 4</span></span>
<span><span class="co">#&gt;    hgnc_id    hgnc_id2 symbol    name                                           </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;                                          </span></span>
<span><span class="co">#&gt;  1 HGNC:11998    11998 TP53      tumor protein p53                              </span></span>
<span><span class="co">#&gt;  2 HGNC:29984    29984 TP53AIP1  tumor protein p53 regulated apoptosis inducingâ€¦</span></span>
<span><span class="co">#&gt;  3 HGNC:11999    11999 TP53BP1   tumor protein p53 binding protein 1            </span></span>
<span><span class="co">#&gt;  4 HGNC:12000    12000 TP53BP2   tumor protein p53 binding protein 2            </span></span>
<span><span class="co">#&gt;  5 HGNC:16328    16328 TP53BP2P1 tumor protein p53 binding protein 2 pseudogeneâ€¦</span></span>
<span><span class="co">#&gt;  6 HGNC:43652    43652 TP53COR1  tumor protein p53 pathway corepressor 1        </span></span>
<span><span class="co">#&gt;  7 HGNC:19373    19373 TP53I3    tumor protein p53 inducible protein 3          </span></span>
<span><span class="co">#&gt;  8 HGNC:16842    16842 TP53I11   tumor protein p53 inducible protein 11         </span></span>
<span><span class="co">#&gt;  9 HGNC:25102    25102 TP53I13   tumor protein p53 inducible protein 13         </span></span>
<span><span class="co">#&gt; 10 HGNC:18022    18022 TP53INP1  tumor protein p53 inducible nuclear protein 1  </span></span>
<span><span class="co">#&gt; # â„¹ 13 more rows</span></span></code></pre></div>
<p>Search for the whole word <code>"TP53"</code> exactly by taking advantage of regular expressions:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hgnc_dataset</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/filter_by_keyword.html">filter_by_keyword</a></span><span class="op">(</span><span class="st">'^TP53$'</span>, cols <span class="op">=</span> <span class="st">'symbol'</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 1 Ã— 4</span></span>
<span><span class="co">#&gt;   hgnc_id    hgnc_id2 symbol name             </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;         &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;            </span></span>
<span><span class="co">#&gt; 1 HGNC:11998    11998 TP53   tumor protein p53</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citing-the-hgnc">Citing the HGNC<a class="anchor" aria-label="anchor" href="#citing-the-hgnc"></a>
</h2>
<p>To cite HGNC nomenclature resources use:</p>
<ul>
<li>Tweedie S, Braschi B, Gray KA, Jones TEM, Seal RL, Yates B, Bruford EA. <em>Genenames.org: the HGNC and VGNC resources in 2021.</em> Nucleic Acids Res. 49, D939â€“D946 (2021). doi: <a href="https://doi.org/10.1093/nar/gkaa980" class="external-link">10.1093/nar/gkaa980</a>
</li>
</ul>
<p>To cite data within the database use the following format:</p>
<ul>
<li>HGNC Database, HUGO Gene Nomenclature Committee (HGNC), European Molecular Biology Laboratory, European Bioinformatics Institute (EMBL-EBI), Wellcome Genome Campus, Hinxton, Cambridge CB10 1SD, United Kingdom www.genenames.org.</li>
</ul>
<p>Please include the month and year you retrieved the data cited.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=hgnc" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/ramiromagno/hgnc/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ramiromagno/hgnc/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing hgnc</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-5226-3441" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Ana-Teresa Maia <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0454-9207" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://cintesis.eu/" class="external-link"><img src="https://rmagno.eu/hgnc/reference/figures/cintesis.svg" height="24"></a> <br><small class="roles"> Copyright holder, funder </small>  </li>
<li>
<a href="https://www.pattern.institute/" class="external-link"><img src="https://rmagno.eu/hgnc/reference/figures/pi-logotype-canonical-whole-v01.svg" height="24"></a> <br><small class="roles"> Copyright holder, funder </small>  </li>
<li>Jacob Munro <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2751-0989" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://CRAN.R-project.org/package=hgnc" class="external-link"><img src="https://www.r-pkg.org/badges/version/hgnc" alt="CRAN status"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://rmagno.eu/hgnc/reference/figures/cintesis.svg" height="24"></a>, <a href="https://www.pattern.institute/" class="external-link"><img src="https://rmagno.eu/hgnc/reference/figures/pi-logotype-canonical-whole-v01.svg" height="24"></a>, Jacob Munro.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
